Dashboard/ViewModels/ConnectionViewModel.cs

```diff
@@ -18,7 +21,9 @@ public partial class ConnectionViewModel : ViewModelBase
     public ObservableCollection<string> SerialPorts { get; }
     private readonly IDataStore? _dataStore;
     public event Action<bool>? ConnectionChanged;
-    public ConnectionViewModel(IDataStore? dataStore = null)
+
+    [ActivatorUtilitiesConstructor]
```

Comment:
What does this do?

Reply:
The ActivatorUtilitiesConstructor tag is to ensure when the ViewModels are created it uses the right constructor since there was a few issues with the test cases not using the right constructor which I only found after removing the Setup page in the TestNavigation test.

Reply:
ohhh very interesting, good find!